Catalog/src/Application/Common/Extensions/ObjectExtensions.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Reflection;
+
+namespace Catalog.Application.Common.Extensions;
+
+public static class ObjectExtensions
```

Comment:
That way any time an entity is extended with additional fields, the mentioned logic needs to be updated. In case of being forgotten, some weird behavior can happen I think.

Reply:
I've refactored a little bit so that cloning is not needed. The first iteration of my code contained sending the whole Product entity for both the "old" and the "updated" value but I changed that to contain only the price some time ago.